### DISCUSSION TITLE
Changing CSS classes

### DISCUSSION THREAD

**Discussion by nobodyinperson** [Textualize/textual#3843] [Q&A] [answered]
Do I understand correctly, that CSS *classes* can't be changed?

So e.g. if I tag some widgets with a `unimplemented` class, what's the best way of toggling their visibility?

Currently, I introduced another class `.hidden { display: none; }`  which I add to or remove from all widgets found by `query(".unimplemented")`. It works but feels a bit complicated.

Is there a way to change a CSS class' rules?

**Comment by davep** [comment, marked as answer]
You can't change the rules of a CSS class; what you would do is create classes with desired rules and than add and remove those classes; which is what you seem to be doing.

**Reply by nobodyinperson** [reply]
Thank you Dave, then I'll just proceed! 😃

**Reply by davep** [reply]
My pleasure.

Was there something particular about what you're doing that made it feel "complicated"? If you give an example perhaps we can suggest the best approach? I ask because normally I find using declarative CSS classes, in combination with `set_class` and the like, especially if combined with [loop-free operations](https://textual.textualize.io/guide/queries/#loop-free-operations), is pretty powerful.

**Reply by nobodyinperson** [reply]
Oooh that is awesome and exactly what I'm looking for. Must have overlooked that section. Thank you!! 👍

**Reply by nobodyinperson** [reply]
Use case is [here](https://gitlab.com/nobodyinperson/git-annex-control-center/-/blob/429b84f07e1bc09be8af9e6047c6102fa143b182/git_annex_control_center/__main__.py#L333-342) btw. I guess that `TabPane`s can't be hidden like this is another topic.

**Reply by davep** [reply]
Sounds like you might want [`TabbedContent.hide_tab`](https://textual.textualize.io/widgets/tabbed_content/#textual.widgets._tabbed_content.TabbedContent.hide_tab), perhaps?

**Reply by nobodyinperson** [reply]
Yep, that's what I use in the link above. I just noticed that setting `display: none` on a `TabPane` doesn't work as expected.

**Reply by davep** [reply]
The thing is that `TabbedContent` is a compound widget, so if you want a tab hiding *and* it's content you really need to go via an API for that.

**Reply by nobodyinperson** [reply]
Understood!
